modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/db/AuthService.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2014-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.db
+
+import java.time.Instant
+
+import cats.effect.{Concurrent, ContextShift}
+
+import cats.implicits._
+
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.sts.StsAsyncClient
+import software.amazon.awssdk.services.sts.model.AssumeRoleRequest
+
+import com.snowplowanalytics.aws.Common
+import com.snowplowanalytics.snowplow.rdbloader.config.StorageTarget
+
+object AuthService {
+  /**
+   * Auth method that is used with COPY INTO statement
+   */
+  sealed trait LoadAuthMethod
+
+  object LoadAuthMethod {
+    /**
+     * Specifies auth method that doesn't use credentials
+     * Destination should be already configured with some other mean
+     * for copying from transformer output bucket
+     */
+    final case object NoCreds extends LoadAuthMethod
+
+    /**
+     * Specifies auth method that pass temporary credentials to COPY INTO statement
+     */
+    final case class TempCreds(awsAccessKey: String, awsSecretKey: String, awsSessionToken: String) extends LoadAuthMethod
+  }
+
+  /**
+   * Get load auth method according to value specified in the config
+   * If temporary credentials method is specified in the config, it will get temporary credentials
+   * with sending request to STS service then return credentials.
+   */
+  def getLoadAuthMethod[F[_]: Concurrent: ContextShift](authMethodConfig: StorageTarget.LoadAuthMethod,
+                                    region: String,
+                                    now: Instant,
+                                    sessionDuration: Integer): F[LoadAuthMethod] =
+    authMethodConfig match {
+      case StorageTarget.LoadAuthMethod.NoCreds => Concurrent[F].pure(LoadAuthMethod.NoCreds)
+      case StorageTarget.LoadAuthMethod.TempCreds(roleArn) =>
+        for {
+          stsAsyncClient <- Concurrent[F].delay(
+              StsAsyncClient.builder()
+                .region(Region.of(region))
+                .build()
+            )
+          assumeRoleRequest <- Concurrent[F].delay(
+              AssumeRoleRequest.builder()
+                .durationSeconds(sessionDuration)
+                .roleArn(roleArn)
+                .roleSessionName(s"rdb_loader_${now.getEpochSecond}")
```

Comment:
I agree with you. I don't have strong opinion on this one. I can remove timestamp from there. Should it be just `rdb_loader` then ?

Reply:
I spoke to Josh.... please can make the role session name completely configurable, with default "rdb_loader"?  In production environments it will be helpful to make it unique to pipeline environment and target.  But that logic doesn't belong in rdb-loader code itself, so we should override the default at run time.